modules/navier_stokes/test/tests/postprocessors/gold/diverging_INSFV_quad_xy_upwind.csv

```diff
@@ -0,0 +1,3 @@
+time,inlet_advected_energy,inlet_mass_constant,inlet_mass_matprop,inlet_mass_variable,inlet_momentum_x,inlet_momentum_y,mid1_advected_energy,mid1_mass,mid1_momentum_x,mid1_momentum_y,mid2_advected_energy,mid2_mass,mid2_momentum_x,mid2_momentum_y,outlet_advected_energy,outlet_mass,outlet_momentum_x,outlet_momentum_y
+0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
+1,-302,-1,-1,-1,-0.024476040741523,-0.94215502182725,310,1,-0.059947284338798,1.0723266687159,310,1,0.029410484806732,0.78663499152553,310,1,0.0034392597619175,0.76467916759833
```

Comment:
I don't understand why `inlet_advected_energy` matches the cell centroid value adjacent to the boundary. I would have thought that the number would be lower because I would think that we are using the two term boundary expansion to extrapolate to the boundary face...and then the upwind ghost value would consequently be lower than the boundary face value because of the gradient.

Reply:
it's a material property here.
With a variable you would be right

Reply:
Oh em gee, I hate the material system more every day. Yea this inherits from `SideFluxIntegral` which is not a `TwoMaterialPropertyInterface` so we only have access to the material property on one side of the face...which makes perfect sense for FE calculations, but doesn't give us the flexibility I would like for finite volume calculations. Yea you have `getADMaterialProperty` which will get the material property at the element centroid, when what we really want for an upwind scheme is `getNeighborADMaterialProperty` which would give us the value on the ghost cell. Could you add this to the issue where we document the need for a more flexible material system? The ticket number is #16809